Clarify distinction between Dashboard widgets

coleshaw opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes #13952

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* No functional changes, just visual changes to clarify role of each column in the Dashboard footer widget.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.

N/A

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to the admin Dashboard
* View the Jetpack footer widget
* See headings for "Jetpack Protect" and "Akismet" (see screenshots for examples).

<img width="621" alt="Screen Shot 2020-02-10 at 7 55 37 PM" src="https://user-images.githubusercontent.com/4930129/74204004-7d79b400-4c3f-11ea-98d2-04c4ed9175ea.png">
<img width="623" alt="Screen Shot 2020-02-10 at 7 56 09 PM" src="https://user-images.githubusercontent.com/4930129/74204010-810d3b00-4c3f-11ea-9271-7a2eae841390.png">



#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* Add headings in Dashboard footer widget. Update copy for consistent verbiage.

coleshaw committed:
Clarify distinction between Dashboard widgets

Fix #13952.

Add headers to clarify that two different features
are being shown in the Dashboard footer widget,
for Jetpack.

jeherve commented on class.jetpack.php:
```diff
@@ -6775,6 +6775,7 @@ public static function dashboard_widget_footer() {
 		<footer>
 
 		<div class="protect">
+			<h4>Jetpack Protect</h4>
```
Could you make this translatable?

coleshaw replied:
Great point, thanks!

jeherve commented on class.jetpack.php:
```diff
@@ -6775,6 +6775,7 @@ public static function dashboard_widget_footer() {
 		<footer>
 
 		<div class="protect">
+			<h4>Jetpack Protect</h4>
```
I wonder if these should be h4, given that we have h3 afterwards to show the numbers. It may be best to keep things hierarchical.

coleshaw replied:
Hm, very true. There is already an `h2` higher on the page, so should these be `h3` / `h4`? That would look like the below:

<img width="252" alt="Screen Shot 2020-02-04 at 8 53 52 PM" src="https://user-images.githubusercontent.com/4930129/73804057-93472f00-4790-11ea-8b56-ffed3b7c6b1e.png">

In this case, the number is less prominent than before -- not sure if that aligns well with the intended goal of the widget? Maybe the number should be a non-header tag, so that it can be styled separately as a larger font?

jeherve replied:
> should these be h3 / h4?

I think we could use h3 for that heading, and not use any heading tag for the number. Instead we could style to look just like before with CSS.

The stylesheet used in that widget is here:
https://github.com/Automattic/jetpack/blob/f2bcdc78dd3ba56b9b07bde603f4d2005464ded6/_inc/client/dashboard-widget/style.scss#L1

jeherve commented on class.jetpack.php:
```diff
@@ -6794,17 +6795,18 @@ public static function dashboard_widget_footer() {
 		</div>
 
 		<div class="akismet">
+			<h4>Akismet</h4>
```
Same as above, we would want this string to be translatable.

jeherve reviewed (changes requested):
Thank you for your contribution. I left some comments below.

coleshaw committed:
Make new headers translatable

coleshaw committed:
Change counts reported from h3 to p with class.

[not verified] (giving error about not escaping the
blocked attempts / spam count)

coleshaw committed:
Style the blocked-count class same as h3 tag.

coleshaw committed:
Superficial whitespace change to returned HTML.

Because Travis CI #43386.6 job timed out (??):

develop.git.wordpress.org[0: 198.143.164.245]: errno=Connection timed out

[not verified]

coleshaw commented:
For reference, here is what the updated widget should look like.

<img width="620" alt="Screen Shot 2020-02-05 at 1 05 59 PM" src="https://user-images.githubusercontent.com/4930129/73872320-7b69bc80-481d-11ea-8f08-780748418b46.png">

keoshi reviewed (changes requested):
Thanks for doing this @coleshaw !

Does that last screenshot reflect the latest changes? If so, could we replace the first one on the original post, please?

This is looking good to me, but I'd like to see two changes before merging:

- Smaller font-size on the product name headings.
- Slightly larger margin-bottom on those same headings.

This should make the product names complementary to the valuable information below: the actual stats and descriptions. I've added a mock on https://github.com/Automattic/jetpack/issues/13952#issuecomment-549321513 :

![image](https://user-images.githubusercontent.com/390760/68118844-f5299b80-fef8-11e9-9f78-57656b151db8.png)

Finally, should we be using the [“Anti-spam” name](https://jetpack.com/upgrade/anti-spam/) instead of Akismet @jeherve ?

jeherve commented:
> should we be using the “Anti-spam” name instead of Akismet

That's a good point, let's do that!

We can take that opportunity to change "Protect" into "Brute force attack protection" as well, to stay consistent with the Jetpack Settings screen.

coleshaw committed:
Change wording to match other product usages.

Protect -> Brute force attack protection
Akismet -> Anti-spam

coleshaw committed:
Make h3 smaller font size in dashboard widget

coleshaw committed:
Add small margin-bottom to h3 in widget.

coleshaw commented:
@keoshi @jeherve , thanks for the feedback. I've updated the code and posted new screenshots in the original PR description. I also updated a couple instances of "Protect" and "Akismet" in the widget, not just in the heading -- do those other changes make sense? Or do you only want the headings to use "Brute force attack protection" and "Anti-spam", and I should leave the other text strings alone?

jeherve commented:
> do those other changes make sense?

I think your changes make sense! 👍

jeherve reviewed (approved):
This works well, even with long strings:
![image](https://user-images.githubusercontent.com/426388/74231215-5d410800-4cc6-11ea-952e-f53a4bd5dc36.png)

keoshi reviewed (approved):
Looks great! Thanks again, @coleshaw !
